application/src/main/java/org/togetherjava/tjbot/commands/bytecode/BytecodeCommand.java

```diff
@@ -0,0 +1,269 @@
+import net.dv8tion.jda.api.events.message.MessageDeleteEvent;
+import net.dv8tion.jda.api.events.message.MessageReceivedEvent;
+import net.dv8tion.jda.api.events.message.MessageUpdateEvent;
+import net.dv8tion.jda.api.interactions.InteractionHook;
+import net.dv8tion.jda.api.interactions.commands.build.Commands;
+import org.jetbrains.annotations.NotNull;
+import org.togetherjava.tjbot.commands.BotCommandAdapter;
+import org.togetherjava.tjbot.commands.CommandVisibility;
+import org.togetherjava.tjbot.commands.MessageContextCommand;
+import org.togetherjava.tjbot.commands.MessageReceiver;
+import org.togetherjava.tjbot.imc.CompilationResult;
+import org.togetherjava.tjbot.imc.CompileInfo;
+import org.togetherjava.tjbot.imc.InMemoryCompiler;
+import org.togetherjava.tjbot.imc.JavacOption;
+import org.togetherjava.tjbot.javap.Javap;
+import org.togetherjava.tjbot.javap.JavapOption;
+
+import javax.tools.Diagnostic;
+import java.util.*;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+
+/**
+ * Bytecode command that uses the message context commands Example usage:
+ * <p>
+ *
+ * <pre>
+ * {@code
+ *     ```java
+ *     class Test {
+ *         public void foo() {
+ *             System.out.println("Hello World!");
+ *         }
+ *     }
+ *     ```
+ * }
+ * </pre>
+ *
+ * then apply the application
+ */
+public final class BytecodeCommand extends BotCommandAdapter
+        implements MessageContextCommand, MessageReceiver {
+    private static final String CODE_BLOCK_OPENING = "```\n";
+    private static final String CODE_BLOCK_CLOSING = "\n```";
+
+    private final Pattern codeBlockExtractorPattern =
+            Pattern.compile("```(?:java)?\\s*([\\w\\W]+)```|``?([\\w\\W]+)``?");
+    private final Map<Long, List<Long>> userToBotMessages = new HashMap<>();
+
+    public BytecodeCommand() {
+        super(Commands.message("View bytecode"), CommandVisibility.GUILD);
+    }
+
+    // Fresh compile when a user uses the message context command on one of their messages
+    @Override
+    public void onMessageContext(@NotNull MessageContextInteractionEvent event) {
+        Message message = event.getTarget();
+        String content = message.getContentRaw();
+
+        event.reply("Compiling...")
+            .mentionRepliedUser(false)
+            .flatMap(InteractionHook::retrieveOriginal)
+            .queue(compReply -> compile(message, compReply, parseCommandFromMessage(content)));
+    }
+
+    // Delete our messages if the user deletes their request message
+    @Override
+    public void onMessageDeleted(@NotNull MessageDeleteEvent event) {
+        long mesageIdLong = event.getMessageIdLong();
+
+        if (userToBotMessages.containsKey(mesageIdLong)) {
+            deleteMyMessages(mesageIdLong, event.getTextChannel());
+        }
+    }
+
+    @Override
+    public void onMessageReceived(@NotNull MessageReceivedEvent event) {
+        /* empty */
```

Comment:
missing the **why**.

Reply:
because it's not needed?

Reply:
the super class is forcing me to implement that method

Reply:
```suggestions
// Method isn't needed
```

Reply:
see, ur implementing an interface that forces everyone to implement `onMessageReceived`. so now u have to explain **why** this method is empty.

Tais suggestion is good. But also just writing `not interested in receiving messages` is okay as well. the point is, writing `empty` doesnt tell anyone why u made it empty.

Reply:
oh I put the comment there without the linter telling me to lmao, it's a bit of mine to put /* empty */ into empty method bodies :p